### PR TITLE
DIG-1557 Data Portal Fixes from runbook on March 27, 2024

### PR DESCRIPTION
**Pull request by OrdiNeu**
## Ticket(s)

-[DIG-1557](https://candig.atlassian.net/browse/DIG-1557)

## Description

Fixes:

- Genomic not displayed while logged in as non-user2?

- Remove one treatment = remove all treatments?

- Switch to two of the same chart, remove a chart in the middle, refresh page, suddenly a different chart is changed

- Edit runbook for genomic testing

- Tumor Registration ID -> Specimen Registration ID

- Add cohort ID to genomic results graph

- Warning: Failed prop type: Invalid prop children of type array supplied to ForwardRef(Tooltip), expected a single ReactElement.

**Does not fix**:

- Switching page size on search page clears the sidebar but doesn't clear the search results

- Sidebar bug if it is open and you reduce page size

## Related Issues (if appropriate)

- [DIG-1563](https://candig.atlassian.net/browse/DIG-1563)

## Types of Change(s)

-   [x] 🪲 Bug fix (non-breaking change that fixes an issue)
-   [ ] ✨ New feature (non-breaking change that adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:

-   [ ] My change requires a change to the documentation
-   [ ] I have updated the documentation accordingly
-   [x] Prettier linter doesn't return errors
-   [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc.
-   [x] Locally tested
-   [ ] Dev server tested
-   [ ] Production tested when merging into stable/production branch
-   [x] [Runbook tasks](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) pass locally/on UHN-Dev
-   [ ] If visuals have changed, [Runbook](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) has been updated with new screenshots

[DIG-1557]: https://candig.atlassian.net/browse/DIG-1557?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DIG-1563]: https://candig.atlassian.net/browse/DIG-1563?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/views/clinicalGenomic/widgets/genomicData.js [hunk 1]
```diff
@@ -49,9 +49,10 @@ function GenomicData() {
     const columns = [
         { field: 'location', headerName: 'Node', minWidth: 120, sortable: false },
         { field: 'donor_id', headerName: 'Donor ID', minWidth: 150, sortable: false },
+        { field: 'program_id', headerName: 'Cohort ID', minWidth: 150, sortable: false },
         { field: 'position', headerName: 'Position', minWidth: 150, sortable: false },
         { field: 'tumour_normal_designation', headerName: 'Tumour/Normal', minWidth: 200, sortable: false },
-        { field: 'submitter_specimen_id', headerName: 'Tumour Specimen ID', minWidth: 300, sortable: false },
+        { field: 'submitter_specimen_id', headerName: 'Specimen Registration ID', minWidth: 300, sortable: false },
```
Comment (mshadbolt) [inline src/views/clinicalGenomic/widgets/genomicData.js]:
```suggestion
        { field: 'submitter_specimen_id', headerName: 'Sample Registration ID', minWidth: 300, sortable: false },
```

**Review by mshadbolt — CHANGES_REQUESTED** [review]
The treatment sidebar selection still seems to be acting weirdly. With the new small synthetic dataset if I select 'hormonal therapy' I get DONOR_1&2, if I then add 'chemotherapy' I get an additional donor DONOR_4, then I remove 'chemotherapy', and I get all donors, 1,2,3,4,5,6

Screenshot 2024-04-19 at 11 49 26 AM
Screenshot 2024-04-19 at 11 49 34 AM
Screenshot 2024-04-19 at 11 49 41 AM

- completeness issue seems fixed
- couldn't reproduce the graph change error
- can we change 'Specimen Registration ID' to 'Sample Registration ID'
- confirm can see genomic completeness per program on graph

**Comment by OrdiNeu** [comment]
> The treatment sidebar selection still seems to be acting weirdly. With the new small synthetic dataset if I select 'hormonal therapy' I get DONOR_1&2, if I then add 'chemotherapy' I get an additional donor DONOR_4, then I remove 'chemotherapy', and I get all donors, 1,2,3,4,5,6

Good catch -- I think I managed to fix it in [8696068](https://github.com/CanDIG/candig-data-portal/pull/135/commits/8696068d53955c522d82054b8b7178533dd77f7d) :heavy_check_mark:

**Comment by mshadbolt** [comment]
Not exactly sure what triggered it. I had previously deleted all the visualisation graphs, then I tried to add the primary_site graph back and this error was thrown and I can't get the search page back, tried refreshing but no dice. I could get the error to go away if I open portal in a new private/incognito window
Screenshot 2024-04-19 at 1 35 24 PM

Maybe it freaks out if I have a filter selected that has no primary sites but try to show a graph of primary sites?

The treatment fix seemed to be working before I did this though. Would it be better for me to make a new ticket for the error above?

**Comment by OrdiNeu** [comment]
@mshadbolt 
> Maybe it freaks out if I have a filter selected that has no primary sites but try to show a graph of primary sites?

I think I got a sort-of fix working in [243b381](https://github.com/CanDIG/candig-data-portal/pull/135/commits/243b381458ea19b20083c1e458aa51a4505281e9), except it won't work if your cache is already messed up. This prevents users from invalidating their localstorage by doing the steps you've described though.

**Review by mshadbolt — APPROVED** [review]
LGTM!

**Pull request merged**